sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,741 @@
+};
+
+// @public (undocumented)
+export type CustomFormModelResponse = CustomFormModel & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: Model;
+    };
+};
+
+// @public (undocumented)
+export type CustomFormModelTrainResult = Omit<TrainResult, "averageModelAccuracy" | "fields">;
+
+// @public
+export class CustomFormRecognizerClient {
+    constructor(endpointUrl: string, credential: TokenCredential | CognitiveKeyCredential, options?: FormRecognizerClientOptions);
+    // (undocumented)
+    deleteModel(modelId: string, options?: DeleteModelOptions): Promise<import("@azure/core-http").RestResponse>;
+    readonly endpointUrl: string;
+    // (undocumented)
+    extractCustomForm(modelId: string, body: HttpRequestBody, contentType: string, options: StartAnalyzePollerOptions<GetAnalyzeFormResultResponse>): Promise<PollerLike<PollOperationState<GetAnalyzeFormResultResponse>, GetAnalyzeFormResultResponse>>;
+    // (undocumented)
+    extractCustomFormFromUrl(modelId: string, imageSourceUrl: string, options: StartAnalyzePollerOptions<GetAnalyzeFormResultResponse>): Promise<PollerLike<PollOperationState<GetAnalyzeFormResultResponse>, GetAnalyzeFormResultResponse>>;
+    // (undocumented)
+    getModel(modelId: string, options: GetModelOptions): Promise<LabeledFormModelResponse | CustomFormModelResponse>;
+    // (undocumented)
+    getSummary(options?: GetSummaryOptions): Promise<GetCustomModelsResponse>;
+    // (undocumented)
+    listModels(options?: ListModelsOptions): PagedAsyncIterableIterator<ModelInfo, GetCustomModelsResponse>;
+    // (undocumented)
+    listModelsAll(settings: PageSettings, options?: ListModelsOptions): AsyncIterableIterator<ModelInfo>;
+    // (undocumented)
+    startTraining(source: string, options?: StartTrainingOptions): Promise<PollerLike<PollOperationState<Model>, Model>>;
+}
+
+// @public
+export interface DataTable {
+    // (undocumented)
+    rows: DataTableRow[];
+}
+
+// @public
+export type DataTableCell = Omit<DataTableCellModel, "elements"> & {
+    elements?: TextElement[];
+};
+
+// @public
+export interface DataTableCellModel {
+    boundingBox: number[];
+    columnIndex: number;
+    columnSpan?: number;
+    confidence: number;
+    elements?: string[];
+    isFooter?: boolean;
+    isHeader?: boolean;
+    rowIndex: number;
+    rowSpan?: number;
+    text: string;
+}
+
+// @public
+export interface DataTableModel {
+    cells: DataTableCellModel[];
+    columns: number;
+    rows: number;
+}
+
+// @public (undocumented)
+export interface DataTableRow {
+    // (undocumented)
+    cells: DataTableCell[];
+}
+
+// @public (undocumented)
+export type DateFieldValue = {
+    type: "date";
+    valueDate: string;
+} & CommonFieldValue;
+
+// @public
+export type DeleteModelOptions = FormRecognizerOperationOptions;
+
+// @public
+export interface DocumentResult {
+    docType: string;
+    fields: {
+        [propertyName: string]: FieldValue_2;
```

Comment:
FieldValue_2 is an unfortunate name, maybe swagger rename one?

Reply:
yes, there's a name conflict. I can re-defined this type

Reply:
they need to have `{ status: OperationStatus }` for the poller to check status.